### PR TITLE
Goodbye, GraphQL

### PR DESCRIPTION
**Pull request by agnivade**
```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
cc @hmhealey @M-ZubairAhmed @larkox @enahum @saturninoabril

**Comment by M-ZubairAhmed** [comment]
Damn 😢

File: server/go.mod [hunk 1]
```diff
@@ -121,6 +119,7 @@ require (
 	github.com/google/uuid v1.3.0 // indirect
 	github.com/gopherjs/gopherjs v1.17.2 // indirect
 	github.com/gorilla/css v1.0.0 // indirect
+	github.com/graph-gophers/graphql-go v1.5.1-0.20230110080634-edea822f558a // indirect
```
Comment (isacikgoz) [inline server/go.mod:122]: Do we still depend on this?

**Review by isacikgoz — COMMENTED** [review]
Just a question 👇

File: server/go.mod (hunk 1 above)
Comment (agnivade) [reply server/go.mod:122]: Yep, seems like it's an indirect dependency now. I did do a `modules-tidy`.

**Review by streamer45 — COMMENTED** [review]
Would this require some cleanup on the client side as well?

**Comment by agnivade** [comment]
Yes, mobile and webapp code needs to be removed. It's left to the other teams.

**Comment by streamer45** [comment]
> Yes, mobile and webapp code needs to be removed. It's left to the other teams.

Sounds good. Would it be safe to be merge this without the client changes? (e.g. accessing the missing feature flag)

**Comment by agnivade** [comment]
Yes it should be fine. The only place where the FF was enabled was in Community and that's disabled now.

**Comment by agnivade** [comment]
@M-ZubairAhmed  @larkox @saturninoabril - Could I get confirmations from each of you that there is no code which relies on the "existence" of the feature flag? i.e. all code checks `ff != nil && ff.graphql == 'true'`. And also please check if there are any other parts of the test infrastructure that could break because of this. I know we had set up nightly e2e tests with graphql enabled, but I think those are removed now. Please check and confirm once. Thanks.

**Comment by M-ZubairAhmed** [comment]
@agnivade On the webapp side everything is behind the flag and it would work just fine provided the flag is turned off. I will check for the test related to this. And let me open a web PR branch with this as the base branch just so we can remove everything at once.

**Comment by agnivade** [comment]
It's not about the flag being turned off. We should be able to handle the flag being not present at all.

**Comment by saturninoabril** [comment]
I can confirm that no related test automation (mattermost/mobile/desktop/calls/playbooks), test management and (QA) test servers that rely on the said feature flag. The nightly test automation run for GraphQL was already removed (scheduled pipeline with the said env variable) and no code change is necessary to revert. There were GraphQL related logics in playbooks E2E tests both at mattermost and playbooks repos but I believe it's on a plugin level and is not affected by this change.

On a quick search, I don't see any Delivery pipeline affected by this change as well.

**Review by isacikgoz — APPROVED** [review]
Thanks for all the fish GraphQL!

**Comment by larkox** [comment]
RE: Breaking if graphQL is not there

On mobile it should not break. Even if the feature flag value is wrong (i.e. true) it will try graphql, fail, and try again with rest. So... All good on that side.

And no tests should be relying on graphQL neither.

**Comment by M-ZubairAhmed** [comment]
> It's not about the flag being turned off. We should be able to handle the flag being not present at all.

Not it would still work considering it same as if the flag is turned off. We check for the condition if the flag is present and then if its turned on. 
Nevertheless i have opened another webapp pr so we can removed all of this in single pr

File: server/channels/app/channel.go [hunk 2]
```diff
@@ -1935,14 +1935,6 @@ func (a *App) GetChannelsForTeamForUser(c request.CTX, teamID string, userID str
 	return a.Srv().getChannelsForTeamForUser(c, teamID, userID, opts)
 }
 
-func (a *App) GetChannelsForTeamForUserWithCursor(c request.CTX, teamID string, userID string, opts *model.ChannelSearchOpts, afterChannelID string) (model.ChannelList, *model.AppError) {
```
Comment (hmhealey) [inline server/channels/app/channel.go:1938]: Do you think we're going to want these cursors in the future at all? I suppose we can always re-add them later on, but these feel like they should be useful even without GraphQL

File: server/public/model/channel.go [hunk 3]
```diff
@@ -201,47 +200,6 @@ func WithID(ID string) ChannelOption {
 	}
 }
 
-// The following are some GraphQL methods necessary to return the
-// data in float64 type. The spec doesn't support 64 bit integers,
-// so we have to pass the data in float64. The _ at the end is
-// a hack to keep the attribute name same in GraphQL schema.
-
-func (o *Channel) CreateAt_() float64 {
```
Comment (hmhealey) [inline server/public/model/channel.go:209]: Are we okay making these changes to the public package outside of a major version? If not, we could always mark them `// Deprecated` and then remove them later

**Review by hmhealey — APPROVED** [review]
I have a couple questions to double check some things, but this looks good to go

File: server/channels/app/channel.go (hunk 2 above)
Comment (agnivade) [reply server/channels/app/channel.go:1938]: If we need them, we can always add them back later. But I don't want unused code lying around.

File: server/public/model/channel.go (hunk 3 above)
Comment (agnivade) [reply server/public/model/channel.go:209]: Yeah good point. But technically, the public module is still in 0.x. So we can do what we want ;)

**Pull request merged**